File: 백준/Bronze/2884. 알람 시계/알람 시계.cc
#include <stdio.h>

void Time(int h, int m, int* resH, int* resM) {
	int total_time = h * 60 + m - 45;

	if (total_time < 0) {
		total_time += 1440;
	}

	*resH = total_time / 60;
	*resM = total_time % 60;
}

int main(void) {
	int h = 0, m = 0;
	int resH = 0, resM = 0;

	if (scanf("%d %d", &h, &m) != 2) return -1;

	Time(h, m, &resH, &resM);

	printf("%d %d", resH, resM);

	return 0;
}

File: 백준/Bronze/2884. 알람 시계/알람 시계.py
import sys
input = sys.stdin.readline

def Time(hour: int, minute: int) -> tuple:
    total_time = ((hour * 60) + minute) - 45
    total_time %= 1440
    
    resH, resM = divmod(total_time, 60)
    
    return resH, resM
   
def solve():
    try:
        line = input().split()
        if not line: return
        
        h, m = map(int, line)
        
        resH, resM = Time(h, m)
        
        print(resH, resM)
    except ValueError:
        pass

if __name__ == "__main__":
    solve()
